### PR TITLE
Lewis

### PR DESCRIPTION
**Pull request by DevRelLewis**
seneca-pipedrive-provider querying deals

### CODE REVIEW

**Comment by rjrodger** [comment]
Give the PR a descriptive title, e.g. "initial implementation"

File: .env.local [hunk 1]
```diff
@@ -0,0 +1,2 @@
+PIPEDRIVE_API_KEY="a4cad8d995eb742a14b27d8faac6287cdd9e3706"
```
Comment (rjrodger) [inline .env.local]: delete this file from the commit
add to .gitignore

**Comment by rjrodger** [comment]
the build is broken: https://github.com/senecajs/seneca-pipedrive-provider/actions/runs/6090944582/job/16579781902

File: .env.production [hunk 2]
```diff
@@ -0,0 +1,2 @@
+PIPEDRIVE_API_KEY=""
```
Comment (rjrodger) [inline .env.production]: this one as well 
these projects don't use .env files for key vars - review webflow-provider

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -1,44 +1,44 @@
...
+    env:
+      WEBFLOW_APIKEY: ${{secrets.WEBFLOW_APIKEY}}
+
+    timeout-minutes: 6
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        node-version: [lts/*, 18.x, 16.x]
```
Comment (rjrodger) [inline .github/workflows/build.yml]: add 18.x and 20.x

File: jest.setup.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const config = {
+  verbose: true,
```
Comment (rjrodger) [inline jest.setup.js]: is there a command line option for this? avoid setting verbose as the default

File: package.json [hunk 5]
```diff
@@ -1,70 +1,73 @@
...
-    "esbuild": "^0.15.7",
-    "esbuild-jest": "^0.5.0",
-    "jest": "^29.0.3",
-    "node-fetch": "^2",
-    "seneca": ">=3",
-    "seneca-msg-test": "^3.4.0",
-    "typescript": "^4.9.4"
-  }
-}
+{
```
Comment (rjrodger) [inline package.json:1]: this file still seems to for webflow?

File: src/WebflowProvider-doc.ts [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
-    desc: 'Get information about the Webflow SDK.',
-  },
-}
-
-export default docs
-
-if ('undefined' !== typeof module) {
-  module.exports = docs
-}
+/* Copyright © 2022 Seneca Project Contributors, MIT License. */
```
Comment (rjrodger) [inline src/WebflowProvider-doc.ts]: delete this file

File: src/pipedrive-provider.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
+const pipedrive = require("pipedrive"); // or however you import pipedrive
```
Comment (rjrodger) [inline src/pipedrive-provider.ts]: seneca code does not have semicolons
see https://standardjs.com/

File: src/pipedrive-provider.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
+const pipedrive = require("pipedrive"); // or however you import pipedrive
+
+const defaultClient = new pipedrive.ApiClient();
+
+// Configure authorization by setting API key
+// PIPEDRIVE_API_KEY is an environment variable that holds the real API key
+defaultClient.authentications.api_key.apiKey =
```
Comment (rjrodger) [inline src/pipedrive-provider.ts]: do not commit api keys

File: src/pipedrive-provider.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
+const pipedrive = require("pipedrive"); // or however you import pipedrive
+
+const defaultClient = new pipedrive.ApiClient();
+
+// Configure authorization by setting API key
+// PIPEDRIVE_API_KEY is an environment variable that holds the real API key
+defaultClient.authentications.api_key.apiKey =
+  "a4cad8d995eb742a14b27d8faac6287cdd9e3706" as string;
+
+export async function getDeals(): Promise<{
```
Comment (rjrodger) [inline src/pipedrive-provider.ts]: this code should look like the webflow provider

File: src/webflow-provider.ts [hunk 10]
```diff
@@ -1,186 +1,186 @@
...
-}
-
-Object.assign(WebflowProvider, { defaults })
-
-export default WebflowProvider
-
-if ('undefined' !== typeof module) {
-  module.exports = WebflowProvider
-}
+/* Copyright © 2022 Seneca Project Contributors, MIT License. */
```
Comment (rjrodger) [inline src/webflow-provider.ts:1]: delete this file

File: test/live.js [hunk 11]
```diff
@@ -1,31 +1,31 @@
...
-  .use('../')
-  .ready(async function () {
-    const seneca = this
-
-    console.log(await seneca.post('sys:provider,provider:webflow,get:info'))
-
-    const list = await seneca.entity('provider/webflow/site').list$()
-    console.log(list.slice(0, 3))
-  })
+const Seneca = require("seneca");
```
Comment (rjrodger) [inline test/live.js]: use this file to test the pipedrive setup

**Review by rjrodger — CHANGES_REQUESTED** [review]

File: .env.local (hunk 1 above)
Comment (DevRelLewis) [reply .env.local]: Deleted from file and the gitignore holds the key in the keys.text file.

File: .env.production (hunk 2 above)
Comment (DevRelLewis) [reply .env.production]: Deleted

For review, you're referring to the webflow-Provider-doc.js file yes?

File: .github/workflows/build.yml (hunk 3 above)
Comment (DevRelLewis) [reply .github/workflows/build.yml]: 18.x is already in the file. 20.x added.

File: jest.setup.js (hunk 4 above)
Comment (DevRelLewis) [reply jest.setup.js]: verbose deleted. 

There is a command for this.

File: package.json (hunk 5 above)
Comment (DevRelLewis) [reply package.json:1]: Can we talk about this? I don't understand. 

Current version is 4.9.5 as of 23 September 2023.

File: src/WebflowProvider-doc.ts (hunk 6 above)
Comment (DevRelLewis) [reply src/WebflowProvider-doc.ts]: File deleted

File: src/pipedrive-provider.ts (hunk 8 above)
Comment (DevRelLewis) [reply src/pipedrive-provider.ts]: The api key was moved during our last session

File: src/pipedrive-provider.ts (hunk 7 above)
Comment (DevRelLewis) [reply src/pipedrive-provider.ts]: Went into VScode settings for prettier and disabled auto add of semicolons. cmd-f for all ";" and deleted them all from the file.

File: src/pipedrive-provider.ts (hunk 8 above)
Comment (DevRelLewis) [reply src/pipedrive-provider.ts]: We also talked about this in our last session and has been addressed. I won't be doing that again.

File: src/webflow-provider.ts (hunk 10 above)
Comment (DevRelLewis) [reply src/webflow-provider.ts:1]: Deleted

File: test/live.js (hunk 11 above)
Comment (DevRelLewis) [reply test/live.js]: Can we run this test together? File has been updated though.

**Pull request merged**